src/Node/Stream.purs

```diff
@@ -74,19 +90,55 @@ onData r cb =
       Right buf ->
         pure buf
 
+read :: forall w eff. Readable w (err :: EXCEPTION | eff) -> Eff (err :: EXCEPTION | eff) (Maybe Buffer)
+read r = do
+  v <- read' r
+  case v of
+       Nothing -> return Nothing
+       Just (BufferData b) -> return (Just b)
+       Just (StringData _) -> throw "Node.Stream.read: Stream encoding should not be set"
```

Comment:
I don't think we need to include `Node.Stream.read` in the error message, as you get stack traces in JS already.